packages/codemods/src/plugins/contract.ts

```diff
@@ -0,0 +1,62 @@
+import { ImportDeclaration, Collection, JSCodeshift } from 'jscodeshift';
+
+// We want to ensure the config contract is correct so devs can get type safety
+type ValidateConfig<T, Struct> = T extends Struct
+  ? Exclude<keyof T, keyof Struct> extends never
+    ? T
+    : never
+  : never;
+
+type BaseConfig = { j: JSCodeshift };
+
+export interface CodemodPlugin {
+  /**
+   * Build the compiled import replacing the existing import
+   *
+   * @param config The configuration object
+   * @param config.j The JSCodeshift object
+   * @param config.currentNode The existing import node that will be replaced
+   * @param config.defaultSpecifierName The import name
+   * @param config.namedImport The export from Compiled to be imported
+   * @param config.compiledImportPath The import path for Compiled
+   *
+   * @returns The import to replace config.currentNode
+   */
+  buildImport?<T>(
+    config: ValidateConfig<
+      T,
+      BaseConfig & {
+        currentNode: ImportDeclaration;
+        defaultSpecifierName: string;
+        namedImport: string;
+        compiledImportPath: string;
+      }
+    >
+  ): ImportDeclaration[];
+
+  /**
+   * Insert AST nodes before the compiled import
+   *
+   * @param config The configuration object
+   * @param config.j The JSCodeshift object
+   * @param config.newImport The new import node replaced in `buildImport`
+   *
+   * @returns Nodes to insert before the import
+   */
+  insertBeforeImport?<T>(
+    config: ValidateConfig<T, BaseConfig & { newImport: Collection<ImportDeclaration> }>
+  ): any;
+
+  /**
+   * Insert AST nodes after the compiled import
+   *
+   * @param config The configuration object
+   * @param config.j The JSCodeshift object
+   * @param config.newImport The new import node replaced in `buildImport`
+   *
+   * @returns Nodes to insert after the import
+   */
+  insertAfterImport?<T>(
```

Comment:
How do you see this working with your initial feature flagging attempt? In Jira's case at least, you want to insert before or after the `styled` or `css` declaration, not the import.

Reply:
This interface doesn't include all the methods required to implement the Jira plugin yet. I was intending to make a follow-up PR with more functionality to reduce the scope of the PR.

My plan would likely be to add hooks before and after the processing of the css and styled blocks, as those will need to be transformed by the codemod to handle the features not implemented.